Apollo/Components/FadeThumb.cs

```diff
@@ -47,44 +67,87 @@ public class FadeThumb: UserControl {
 
         bool dragged = false;
 
-        void DragStarted(object sender, VectorEventArgs e) {
+        void DragStarted(object sender, VectorEventArgs e)
+        {
             ((Window)this.GetVisualRoot()).Focus();
             dragged = false;
         }
 
-        void DragCompleted(object sender, VectorEventArgs e) {
+        void DragCompleted(object sender, VectorEventArgs e)
+        {
             if (!dragged) Focused?.Invoke(this);
             else if (change != 0) Moved?.Invoke(this, 0, change);
         }
 
-        void MouseDown(object sender, PointerPressedEventArgs e) {
+        void MouseDown(object sender, PointerPressedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
             if (MouseButton != PointerUpdateKind.LeftButtonPressed) e.Handled = true;
-            
+
             ((Window)this.GetVisualRoot()).Focus();
         }
 
-        void MouseUp(object sender, PointerReleasedEventArgs e) {
+        void MouseUp(object sender, PointerReleasedEventArgs e)
+        {
             PointerUpdateKind MouseButton = e.GetCurrentPoint(this).Properties.PointerUpdateKind;
 
-            if (MouseButton == PointerUpdateKind.RightButtonReleased) {
-                Deleted?.Invoke(this);
+            if (MouseButton == PointerUpdateKind.RightButtonReleased)
+            {
+                MenuItem selectedItem = null;
+                foreach (Control item in ThumbContextMenu.Items)
+                {
+                    if (item.GetType() != typeof(MenuItem)) break;
```

Comment:
you can do `if (!(item is MenuItem menuItem)) break;` here and you'll have a menuItem var accessible even if the check fails